astropy/visualization/wcsaxes/coordinate_helpers.py

```diff
@@ -186,11 +186,12 @@ def grid(self, draw_grid=True, grid_type=None, **kwargs):
 
         self.grid_lines_kwargs.update(kwargs)
 
-        if self.grid_lines_kwargs["visible"]:
-            if not draw_grid:
-                self.grid_lines_kwargs["visible"] = False
-        else:
+        if isinstance(draw_grid, bool):
+            self.grid_lines_kwargs["visible"] = draw_grid
+        elif draw_grid is None:
             self.grid_lines_kwargs["visible"] = True
+        else:
```

Comment:
This `else` has no test coverage. In practice, what would user actually do to trigger this? Does matplotlib allow something else that we are not supporting or something? Regardless, a test to make sure this exception gets thrown as designed would be nice. Thanks!

Reply:
Good point ! We could actually simplify the tree again so this else isn't needed. Something like
```python
if draw_grid is None:
    draw_grid = True
self.grid_lines_kwargs["visible"] = draw_grid
```

Reply:
I have changed the patch and also removed the test for the 'else' part